src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -2667,12 +2667,12 @@ internal void GetReferenceItems
                 }
             }
 
-            primaryFiles = primaryItems.ToArray();
-            dependencyFiles = dependencyItems.ToArray();
-            relatedFiles = relatedItems.ToArray();
-            satelliteFiles = satelliteItems.ToArray();
-            serializationAssemblyFiles = serializationAssemblyItems.ToArray();
-            scatterFiles = scatterItems.ToArray();
+            primaryFiles = NormalizePathsAsArray(primaryItems);
+            dependencyFiles = NormalizePathsAsArray(dependencyItems);
+            relatedFiles = NormalizePathsAsArray(relatedItems);
+            satelliteFiles = NormalizePathsAsArray(satelliteItems);
```

Comment:
Look at how hard it would be to move the normalization to the input layer (so that it gets put in the cache).

Reply:
After looking at this bit, I don't think this is an easy change because in addition to finding canonical forms for all input paths, we would have to find canonical forms for dependencies, and that would get complicated.

Reply:
I don't think you should normalize _task_ inputs, but the inputs to the cache--the outputs of the task, but at the layer where they're created, not at output time.

Reply:
Ah, that clarifies things, thank you. Now I'm back to "seems like a good move, but I have no idea how hard that is."

Reply:
Ok, I _think_ the newest changes should be equivalent, but I'm not at all confident.

It seems like the only way anything is added to References is via that one AddReference method. I'm also looking to see if I can move it any earlier. At worst, this is cleaner.

Reply:
Important part of the change is:
https://github.com/dotnet/msbuild/pull/6533/commits/8ed6c4bdde868c1e24be3b3c279ac633a51d07bf